src/navigation/_AppSettingsProvider.jsx

```diff
@@ -20,6 +20,15 @@ const AppSettingsProvider = ({ settingsConfig, children }) => {
     themes,
   } = settingsConfig;
 
+  /* eslint-disable no-undef, no-console, no-param-reassign */
```

Comment:
We should be able to rework the code to remove most of these eslint disables.

To remove the no-undef try declaring the global at the top of the file. Here's an example:

https://github.com/cerner/terra-application/blob/master/src/application-base/private/getBrowserLocale.js#L1

To remove the no-param-reassign try extracting the local out and assigning it as necessary. 

```jsx
let locale = settingsConfig.locale;

// Check if the global exists before using it. Otherwise it will throw an error when it's not defined.
if (typeof TERRA_AGGREGATED_LOCALES === 'object') {
  locales = TERRA_AGGREGATED_LOCALES;
}
```

You could also use a ternary here.

```jsx
const locales = typeof TERRA_AGGREGATED_LOCALES === 'object' ? TERRA_AGGREGATED_LOCALES : settingsConfig.locales;
```

And then inside of the memo config spread locales:

```jsx
    return ({
      ...settingsConfig,
      locales, // Spreading it here will overwrite anything in settingsConfig
      currentLocale,
      currentTheme,
      currentDirection,
      currentThemeName: themes[currentTheme],
      onUpdate,
    });
```

Feel free to experiment with any variations of these examples or something different you come up with, focus on restructuring the code to honor the lint configurations. Whenever you disable a lint rule there is often a different approach that can be made for the code change.